scripts/quests/bastok/Out_of_the_Depths.lua

```diff
@@ -0,0 +1,225 @@
+-----------------------------------
+-- Out of the Depths
+-----------------------------------
+-- Log ID    : 1, Quest ID: 75
+-- Ayame     : !pos 133 -19 34 237
+-- Ravorara  : !pos -151 -7 -7 236
+-- Brakobrik : !pos 164 11 -91 11
+-- Pavvke    : !pos 7.5 7 -13.5 234
+-----------------------------------
+local ID = zones[xi.zone.BASTOK_MARKETS]
+-----------------------------------
+local quest = Quest:new(xi.questLog.BASTOK, xi.quest.id.bastok.OUT_OF_THE_DEPTHS)
+
+quest.reward =
+{
+    fame = 80,
+    fameArea = xi.fameArea.BASTOK,
+    title = xi.title.TRASH_COLLECTOR,
+    gil = 1200,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:getFameLevel(xi.fameArea.BASTOK) >= 3
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Ayame'] = quest:progressEvent(859),
+
+            onEventFinish =
+            {
+                [859] = function(player, csid, option, npc)
+                    if option == 3 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Ayame'] = quest:event(860)
+        },
+
+        [xi.zone.PORT_BASTOK] =
+        {
+            ['Ravorara'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') >= 1 then
+                        if player:hasKeyItem(xi.ki.DUSTY_TOME) then
+                            return quest:progressEvent(309, 0, 1, 0, 597)
+                        elseif player:hasKeyItem(xi.ki.POINTED_JUG) then
+                            return quest:progressEvent(309, 0, 2, 0, 598)
+                        elseif player:hasKeyItem(xi.ki.CRACKED_CLUB) then
+                            return quest:progressEvent(309, 0, 3, 0, 599)
+                        elseif player:hasKeyItem(xi.ki.PEELING_HAIRPIN) then
+                            return quest:progressEvent(309, 0, 4, 0, 600)
+                        elseif player:hasKeyItem(xi.ki.OLD_NAMETAG) then
+                            return quest:progressEvent(309, 0, 5, 0, 601)
+                        else
+                            return quest:progressEvent(308, 0)
```

Comment:
Ok - I was under the impression that, while it would "work" we wanted to keep progressEvent to just that, progressing the event. If its a CS or reminder that didn't move it forward, then use event and thus no onEventFinish.